cyclonedx/model/issue.py

```diff
@@ -263,3 +274,17 @@ def set_source_url(self, source_url: XsUri) -> None:
             self._source.url = source_url
         else:
             self._source = IssueTypeSource(url=source_url)
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, IssueType):
+            return hash(other) == hash(self)
+        return False
+
+    def __hash__(self) -> int:
+        return hash((
+            self._type, self._id, self._name, self._description, self._source,
+            tuple([hash(ref) for ref in set(sorted(self._references, key=hash))]) if self._references else None
```

Comment:
no need for the list-constructor `[]` inside the tuple constructor `tuple()`
the tuple constructor accepts iterable as an constructor property.

and then the `set` sinstructor around the `sorted` is not needed. `sorted` returns a pre-sorted object, `set` does not support orders at all.

and then the `hash()` call is not needed, as the outer `hash()` will call `hash()` on the inner elements if needed.

the following should be enough
```patch
- tuple([hash(ref) for ref in set(sorted(self._references, key=hash))])
+ tuple(sorted(self._references, key=hash))
```

when looking at the code i see a huge design opportunity: 
all the element lists (like `references` etc ...) are actually intended to be not `list`but `set` in the first place.
from the spec the order of the elements does not matter at all. 

:1st_place_medal:  conclusion: 
* all the optional lists should be  sets. 
  *  therefore the internal property would be a set at all times
  * the getter would return `set` at all times
  * the parameters in the getter and constructor would accept not just sets but all implementations of `Iterable`
* do not call unnecessary `hash()` on internal implementations

Reply:
Agree with both points.

I'll initially address the `hash()` calls in this PR.

Have already raised #159 regarding use of `Set` vs `List`. Will be a big code change and just weighing up time and whether to hold back `2.0.0` for this or not?

Reply:
Gah - simplifying the hash code as you call out basically requires us to migrate to Sets:

```
    tuple(sorted(self.external_references, key=hash)), tuple(sorted(self.properties, key=hash)),
TypeError: 'NoneType' object is not iterable
```